pkg/network/node/ovs/parse.go

```diff
@@ -397,3 +414,131 @@ func UnparseExternalIDs(externalIDs map[string]string) string {
 	}
 	return "{" + strings.Join(ids, ",") + "}"
 }
+
+func ParseGroup(group string) (*OVSGroup, error) {
+	// According to the man page "Such flow descriptions comprise a series  field=value
+	// assignments,  separated  by  commas or white space.  (Embedding spaces into a
+	// group description normally requires quoting to prevent the  shell  from breaking
+	// the description into multiple arguments.)."
+	parsed := &OVSGroup{}
+
+	origGroup := group
+	for group != "" {
+		field := ""
+		value := ""
+		end := strings.IndexAny(group, ", ")
+		if end == -1 {
+			end = len(group)
+		}
+		eq := strings.Index(group, "=")
+		if eq == -1 || eq > end {
+			field = group[:end]
+		} else {
+			field = group[:eq]
+			if field == "bucket" {
+				//find the index of the next bucket to determine the end
+				end = strings.Index(group[eq+1:], "bucket")
+				if end == -1 {
+					end = len(group)
+				} else {
+					value = group[eq+1 : eq+1+end]
+					end = eq + 1 + end
+				}
+			}
+			valueEnd := end - 1
+			for group[valueEnd] == ' ' || group[valueEnd] == ',' {
+				valueEnd--
+			}
+			value = strings.Trim(group[eq+1:end], ", ")
+			if value == "" {
+				return nil, fmt.Errorf("bad group definition %q (empty field %q)", origGroup, field)
+			}
+		}
+
+		switch field {
+		case "group_id":
+			converted, err := strconv.ParseUint(value, 10, 32)
+			if err != nil {
+				return nil, fmt.Errorf("Cannot convert %s to group_id: %v", value, err)
+			}
+			parsed.GroupID = uint32(converted)
+		case "type":
+			if value == "select" {
+				parsed.Type = SelectGroup
+			} else {
+				return nil, fmt.Errorf("groups of type %s are not supported", value)
+			}
+		case "bucket":
+			//remove actions either hear or above
```

Comment:
```suggestion
			// remove actions either here or above
```
I don't understand what this means though...

Reply:
left a message to myself while I was coding it. removed the message